components/CalEPD/include/gdeh0154z90.h

```diff
@@ -0,0 +1,55 @@
+// Contributed by thebino: https://github.com/martinberlin/cale-idf/issues/47
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "freertos/FreeRTOS.h"
+#include "esp_system.h"
+#include <stdint.h>
+#include <math.h>
+#include "sdkconfig.h"
+#include "esp_log.h"
+#include <string>
+#include <epd.h>
+#include <Adafruit_GFX.h>
+#include <epdspi.h>
+#include <gdew_colors.h>
+
+// Controller: SSD1681  Note: Used by Waveshare in https://www.waveshare.com/wiki/1.54inch_e-Paper_Module_(B)
+#define GDEH0154Z90_WIDTH 200
+#define GDEH0154Z90_HEIGHT 200
+#define GDEH0154Z90_BUFFER_SIZE (uint32_t(GDEH0154Z90_WIDTH) * uint32_t(GDEH0154Z90_HEIGHT) / 8)
+
+#define GDEH0154Z90_8PIX_BLACK 0x00
+#define GDEH0154Z90_8PIX_WHITE 0xFF
+#define GDEH0154Z90_8PIX_RED 0xFF
+#define GDEH0154Z90_8PIX_RED_WHITE 0x00
+
+class Gdeh0154z90 : public Epd
+{
+public:
+    Gdeh0154z90(EpdSpi &IO);
+    uint8_t colors_supported = 2;
+    uint8_t partial_update_supported = 0;
+
+    void init(bool debug);
+    void fillScreen(uint16_t color);
+    void drawPixel(int16_t x, int16_t y, uint16_t color); // Override GFX own drawPixel method
+
+    void update();
+    void updateWindow(uint16_t x, uint16_t y, uint16_t w, uint16_t h, bool using_rotation = true);
+    void updateToWindow(uint16_t xs, uint16_t ys, uint16_t xd, uint16_t yd, uint16_t w, uint16_t h, bool using_rotation = true);
```

Comment:
updateToWindow -> can be removed completely

Reply:
I added it in preparation of collating a general interface used by the Epd class.